src/shortestpaths/yen.jl

```diff
@@ -0,0 +1,102 @@
+"""
+    struct YenState{T, U}
+Designed for yen k-shortest-paths calculations.
+"""
+struct YenState{T, U<:Integer}<: AbstractPathState
+    dists::Vector{T}
+    paths::Vector{Vector{U}}
+end
+
+
+"""
+  yen_k_shortest_paths(g, source, target, distmx=DefaultDistance(), K);
+
+  Perform [Yen's algorithm](http://en.wikipedia.org/wiki/Yen%27s_algorithm)
+  on a graph, computing k-shortest distances between `source` and `target` other vertices.
+  Return a [`YenState`](@ref) that contains distances and paths.
+
+  ### Optional Arguments
+  - `k=1`: The number of paths to find.
+
+"""
+function yen_k_shortest_paths(
+    g::AbstractGraph,
+    source::U,
+    target::U,
+    distmx::AbstractMatrix{T}=DefaultDistance(),
+    K::Int=1) where T where U<:Integer
+
+    if source == target
+        return YenState{T, U}([U(0)], [[source]])
+    end
+
+    dj = dijkstra_shortest_paths(g, source, distmx)
+    path = enumerate_paths(dj)[target]
+    if isempty(path)
+        return YenState{T, U}(Vector{T}(), Vector{Vector{U}}())
+    end
+
+    dists = Array{T,1}()
+    push!(dists,dj.dists[target])
+    A = [path]
+    B = PriorityQueue()
+    gc = deepcopy(g)
```

Comment:
Perhaps rename this variable to something else? Julia defines the garbage collector as `gc` already.